apps/root/src/common/utils/transaction-history/parsing.tsx

```diff
@@ -592,4 +596,336 @@ const parseMultipleTransactionApiEventsToTransactionEvents = (
   );
 };
 
-export default parseMultipleTransactionApiEventsToTransactionEvents;
+const buildBaseDcaPendingEventData = (position: Position): BaseDcaDataEvent => {
+  const fromToken = { ...position.from, icon: <TokenIcon size={5} token={position.from} /> };
+  const toToken = { ...position.to, icon: <TokenIcon size={5} token={position.to} /> };
+  const positionId = Number(position.positionId);
+  const hub = HUB_ADDRESS[position.version][position.chainId];
+
+  return {
+    fromToken,
+    toToken,
+    positionId,
+    hub,
+  };
+};
+
+export const transformNonIndexedEvents = (
+  events: TransactionDetails[],
+  userWallets: string[],
+  tokenList: TokenList
+): TransactionEvent[] => {
+  if (!events) return [];
+  const eventsPromises = events.map<TransactionEvent | null>((event) => {
+    const network = find(NETWORKS, { chainId: event.chainId }) as NetworkStruct;
+
+    const { nativeCurrencyToken, mainCurrencyToken } = getNetworkCurrencyTokens(network);
+
+    const protocolToken = getProtocolToken(event.chainId);
+    const baseEvent = {
+      tx: {
+        network: {
+          ...network,
+          nativeCurrency: {
+            ...nativeCurrencyToken,
+            icon: <TokenIcon size={5} token={nativeCurrencyToken} />,
+          },
+          mainCurrency: { ...mainCurrencyToken, icon: <TokenIcon size={5} token={mainCurrencyToken} /> },
+        },
+        chainId: event.chainId,
+        txHash: event.hash as Address,
+        timestamp: event.addedTime,
+        explorerLink: buildEtherscanTransaction(event.hash, event.chainId),
+        initiatedBy: event.from as Address,
+        spentInGas: {
+          amount: event.receipt?.gasUsed || 0n,
+          amountInUnits: formatCurrencyAmount(event.receipt?.gasUsed || 0n, nativeCurrencyToken),
+        },
+        nativePrice: parseFloat(formatUnits(event.receipt?.effectiveGasPrice || 0n, nativeCurrencyToken.decimals)),
```

Comment:
wait this is not the nativePrice. effectiveGasPrice is to calculate how much ETH was spent on the transaction. so actually spentInGas amount is effectiveGasPrice * gasUsed

Reply:
We should try to get the native price from the balance state